src/snowflake/cli/plugins/git/commands.py

```diff
@@ -38,6 +42,79 @@ def _repo_path_argument_callback(path):
 )
 
 
+def _assure_repository_does_not_exist(repository_name: str) -> None:
+    om = ObjectManager()
+    try:
+        om.describe(
+            object_type=ObjectType.GIT_REPOSITORY.value.cli_name,
+            name=repository_name,
+        )
+        raise ClickException(f"Repository '{repository_name}' already exists")
+    except ProgrammingError:
+        pass
+
+
+@app.command("setup", requires_connection=True)
+def setup(
+    repository_name: str = RepoNameArgument,
+    **options,
+) -> CommandResult:
+    """
+    Sets up a git repository object.
+
+    You will be prompted for:
+
+    * url - address of repository to be used for git clone operation
+
+    * secret - Snowflake secret containing authentication credentials. Not needed if origin repository does not require
+    authentication for RO operations (clone, fetch)
+
+    * API integration - object allowing Snowflake to interact with git repository.
+    """
+    _assure_repository_does_not_exist(repository_name)
+    manager = GitManager()
+
+    url = typer.prompt("Origin url")
```

Comment:
Should we validate if not empty?

Reply:
In that case typer asks again:
```
▶ snow git setup repo
Origin url: 
Origin url:  
Origin url: 
Origin url: 
```
I was considering validating whether it starts with `https://`, wdyt?

Reply:
+1 for at least https validation

Reply:
done